app/src/test/java/org/oppia/app/testing/ProfileChooserSpanTest.kt

```diff
@@ -0,0 +1,147 @@
+package org.oppia.app.testing
+
+import android.content.Context
+import android.content.res.Configuration
+import android.view.View
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.core.app.ActivityScenario.launch
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.intent.Intents
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.profile.ProfileChooserFragment
+import org.robolectric.annotation.Config
+
+private const val TAG_PROFILE_CHOOSER_FRAGMENT_RECYCLERVIEW = "profile_recycler_view"
+
+/**
+ * Tests for ensuring [ProfileChooserFragment] uses the correct column count for profiles based on display density.
+ * document reffered :https://developer.android.com/reference/androidx/test/core/app/ActivityScenario
+ * */
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class ProfileChooserSpanTest {
+
+  private lateinit var context: Context
+  private var activity: ProfileChooserFragmentTestActivity? = null
+  private lateinit var fragment: ProfileChooserFragment
+  private lateinit var recyclerView: RecyclerView
+
+  @Before
+  @ExperimentalCoroutinesApi
+  fun setUp() {
+    Intents.init()
+    activity = getProfileChooserTestActivity()
+    fragment = getProfileChooserFragment()
+    recyclerView = (fragment!!.view!!.findViewWithTag<View>(TAG_PROFILE_CHOOSER_FRAGMENT_RECYCLERVIEW) as RecyclerView)
+    ApplicationProvider.getApplicationContext<Context>().resources.configuration.orientation =
+      Configuration.ORIENTATION_LANDSCAPE
+    context = ApplicationProvider.getApplicationContext()
+  }
+
+  @get:Rule
+  var activityRule = ActivityScenarioRule(ProfileChooserFragmentTestActivity::class.java)
+
+  private fun getProfileChooserTestActivity(): ProfileChooserFragmentTestActivity? {
+    activityRule.scenario.onActivity {
+      activity = it
+    }
+    return activity as ProfileChooserFragmentTestActivity
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  @Test
+  @Config(qualifiers = "ldpi")
+  fun testProfileChooserSpanTest_onConfigLandScapeAndScreenldpi() {
```

Comment:
These dimension tests can be removed now that you have ``spanCount`` versions of them.

Reply:
Done.